File: VB/CustomCheckComboBox/CustomComboBoxEditSettings.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Xpf.Editors.Settings
Imports DevExpress.Xpf.Editors.Helpers

Namespace CustomCheckComboBox
	Public Class CustomComboBoxEditSettings
		Inherits ComboBoxEditSettings

		Shared Sub New()
			RegisterCustomEdit()
		End Sub
		Protected Overrides Function CreateItemsProvider() As ItemsProvider
			Return New ItemsIndexFinder(Me)
		End Function
		Public Shared Sub RegisterCustomEdit()
			EditorSettingsProvider.Default.RegisterUserEditor(GetType(CustomComboBoxEdit), GetType(CustomComboBoxEditSettings), Function() New CustomComboBoxEdit(), Function() New CustomComboBoxEditSettings())
		End Sub
	End Class
End Namespace

File: CS/CustomCheckComboBox/CustomComboBoxEditSettings.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Xpf.Editors.Settings;
using DevExpress.Xpf.Editors.Helpers;

namespace CustomCheckComboBox {
    public class CustomComboBoxEditSettings : ComboBoxEditSettings {

        static CustomComboBoxEditSettings() {
            RegisterCustomEdit();
        }
        protected override ItemsProvider CreateItemsProvider() {
            return new ItemsIndexFinder(this);
        }
        public static void RegisterCustomEdit() {
            EditorSettingsProvider.Default.RegisterUserEditor(typeof(CustomComboBoxEdit),
                typeof(CustomComboBoxEditSettings),
                () => new CustomComboBoxEdit(),
                () => new CustomComboBoxEditSettings());
        }
    }
}
